From 933047394e7c6a0737527263c9bb1a91e1e93781 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sun, 3 Jul 2016 16:54:35 +0100
Subject: [PATCH] Revert "raven: Restore lightdm (only) screen locking"

This reverts commit 9392146c07b93d8b90681dafbd74d03e15590f12.
We now use gnome-screensaver and not the lightdm locking functionality.
---
 raven/powerstrip.vala | 18 +++++++++---------
 1 file changed, 9 insertions(+), 9 deletions(-)

diff --git a/raven/powerstrip.vala b/raven/powerstrip.vala
index 627b234..56031bc 100644
--- a/raven/powerstrip.vala
+++ b/raven/powerstrip.vala
@@ -12,23 +12,24 @@
 namespace Budgie
 {
 
-[DBus (name="org.gnome.SessionManager")]
-public interface SessionManager : Object
+
+[DBus (name="org.gnome.ScreenSaver")]
+public interface ScreenSaver : Object
 {
-    public abstract async void Shutdown() throws Error;
+    public abstract void lock() throws Error;
 }
 
-[DBus (name="org.freedesktop.DisplayManager.Seat")]
-public interface DMSeat : Object
+[DBus (name="org.gnome.SessionManager")]
+public interface SessionManager : Object
 {
-    public abstract void lock() throws IOError;
+    public abstract async void Shutdown() throws Error;
 }
 
 
 class PowerStrip : Gtk.EventBox
 {
 
-    private DMSeat? saver = null;
+    private ScreenSaver? saver = null;
     private SessionManager? session = null;
 
     private Gtk.Button? lock_btn = null;
@@ -36,9 +37,8 @@ class PowerStrip : Gtk.EventBox
 
     async void setup_dbus()
     {
-        var path = Environment.get_variable("XDG_SEAT_PATH");
         try {
-            saver = yield Bus.get_proxy(BusType.SYSTEM, "org.freedesktop.DisplayManager", path);
+            saver = yield Bus.get_proxy(BusType.SESSION, "org.gnome.ScreenSaver", "/org/gnome/ScreenSaver");
         } catch (Error e) {
             warning("Unable to contact login manager: %s", e.message);
             return;
-- 
2.9.0

